internal/telemetry/region_pusher_test.go

```diff
@@ -22,142 +22,194 @@ const (
 	regionID = 1
 )
 
-var (
-	ee = []Execution{
-		{
-			LocalTenantID: 1,
-			CheckClass:    sm.CheckClass_PROTOCOL,
-			Duration:      59 * time.Second,
-		},
-		{
-			LocalTenantID: 1,
-			CheckClass:    sm.CheckClass_PROTOCOL,
-			Duration:      60 * time.Second,
-		},
-		{
-			LocalTenantID: 2,
-			CheckClass:    sm.CheckClass_SCRIPTED,
-			Duration:      61 * time.Second,
-		},
-		{
-			LocalTenantID: 2,
-			CheckClass:    sm.CheckClass_SCRIPTED,
-			Duration:      30 * time.Second,
-		},
-
-		{
-			LocalTenantID: 1,
-			CheckClass:    sm.CheckClass_SCRIPTED,
-			Duration:      30 * time.Second,
-		},
-		{
-			LocalTenantID: 1,
-			CheckClass:    sm.CheckClass_SCRIPTED,
-			Duration:      59 * time.Second,
-		},
-		{
-			LocalTenantID: 1,
-			CheckClass:    sm.CheckClass_PROTOCOL,
-			Duration:      130 * time.Second,
-		},
-		{
-			LocalTenantID: 1,
-			CheckClass:    sm.CheckClass_SCRIPTED,
-			Duration:      60 * time.Second,
-		},
-		{
-			LocalTenantID: 2,
-			CheckClass:    sm.CheckClass_PROTOCOL,
-			Duration:      45 * time.Second,
-		},
-		{
-			LocalTenantID: 1,
-			CheckClass:    sm.CheckClass_SCRIPTED,
-			Duration:      65 * time.Second,
-		},
-	}
+type testData struct {
+	executions []Execution
+	message    sm.RegionTelemetry
+}
 
-	mm = []sm.RegionTelemetry{
-		// 0: this is the expected message for ee[0] -> ee[3] executions
+func getTestDataset(idx int) testData {
```

Comment:
I ended up doing this because at some point it felt like I was fighting a water balloon: I would squeeze here and it would blow up over there.